docs/utilities/config_annotations.md

```diff
@@ -0,0 +1,234 @@
+
+    It is strongly recommended that you do not use the newline character `\n` (or platform-specific equivalents) in comments, and instead use the multi-line `@Comment` annotation form, as Forge will then be able to ensure the proper handling of the multi-line comment.
+
+Property Ranges
+-------------------------
+
+While when using boolean configuration properties all possible input values are considered valid, in the case of numeric inputs there may arise the need as to constrain the acceptable inputs to a certain continuous range. For this reason, the annotations `@RangeInt` and `@RangeDouble` exist.
+
+The `@Range` annotations consist of two values, a minimum value and a maximum value, which together define an inclusive range of acceptable inputs for a given numeric property. Note, that inclusive means that the values set for `min` and `max` are considered valid inputs for the property. If either min or max are not set, then the annotation will default to `<? extends Number>.MIN_VALUE` and `<? extends Number>.MAX_VALUE` respectively.
+
+Building off the prior examples a sample usage of the `@RangeInt` and `@RangeDouble` would be as follows.
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This integer property accepts values 1 to 10 inclusive.")
+  @RangeInt(min = 1, max = 10)
+  public static int myIntegerProperty = false;
+
+  @Comment("This double property accepts values 0.0 to 1.0 inclusive.")
+  @RangeDouble(min = 0.0, max = 1.0)
+  public static double myDoubleProperty = false;
+
+}
+````
+
+!! important
+
+    While both the `@RangeInt` and `@RangeDouble` accept a min and a max value to be set they are not interchangeable, given that the type must match. Consequently the `@RangeInt` annotation is only to be applied to `int` and `Integer` fields, and the `@RangeDouble` is only to be applied to `double` and `Double` fields.
+
+!! important
+
+    There currently exists no range annotations capable of supporting numeric types other than the aforementioned `int`, `Integer`, `double`, and `Double` types. As such, you are forced to use `int` instead of `byte` or `long`, `Integer` instead of `Byte` or `Long`, `double` instead of `float`, and `Double` instead of `Float`. This issue will likely be fixed in a future release given wide enough adoption of the new annotation-based configuration system.
+
+
+Sub-categories
+-------------------------
+
+Smaller mods will generally have fewer properties (say about 5 - 10) and do not need to worry about dividing up their configurations into more manageable groups of properties for the end-user to manage. However, for larger mods configurations consisting of 20 to 50 elements is not all that uncommon, and the need for sub-categories becomes irrefutable. Recognizing such a need, the annotation-based configuration system provides a mechanism by which `@Config` classes can be divided into groups that are reflected directly in the structure of the `@Config` class.
```

Comment:
The comparison of small/large mods is totally unnecessary here. People should be able to analyze whether they need subcategories on their own based on their own needs.